pkg/pool-gyro/contracts/Gyro2CLPPool.sol

```diff
@@ -90,16 +90,24 @@ contract Gyro2CLPPool is IGyro2CLPPool, BalancerPoolToken {
         // New invariant
         invariant = invariant.mulUp(invariantRatio);
         uint256 squareNewInv = invariant * invariant;
-        // L / sqrt(beta)
-        uint256 a = invariant.divDown(sqrtBeta);
-        // L * sqrt(alpha)
-        uint256 b = invariant.mulDown(sqrtAlpha);
 
         if (tokenInIndex == 0) {
             // if newBalance = newX
+
+            // L / sqrt(beta), rounded down to minimize newBalance.
+            uint256 a = invariant.divDown(sqrtBeta);
+            // L * sqrt(alpha), rounded up to minimize newBalance (b is in the denominator).
+            uint256 b = invariant.mulUp(sqrtAlpha);
+
             newBalance = squareNewInv.divUpRaw(b + balancesLiveScaled18[1]) - a;
```

Comment:
`a` rounded down is fine, but `b` should be rounded down too. We always want to round `newBalance` up, and b is n the denominator so we want a smaller `b`.

Reply:
Ooooff, I inverted the logic. Sure, we should always round newBalance up.